File: app/src/main/java/com/example/template/app/BaseActivity.java
package com.example.template.app;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * 功能描述：
 * Created by gfq on 2019/12/20.
 */
public abstract class BaseActivity extends AppCompatActivity {

    protected BaseActivity mActivity;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mActivity = this;
    }

}

File: appkotlin/src/main/java/com/example/appkotlin/base/BaseActivity.kt
package com.example.appkotlin.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 *
 * 功能描述：
 * Created by gfq on 2020/12/1
 *
 **/

abstract class BaseActivity : AppCompatActivity() {

    lateinit var mActivity: BaseActivity

    abstract fun getLayoutId(): Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
        mActivity = this
    }
}